specs/tls.rst

```diff
@@ -0,0 +1,201 @@
+  it will be ``None``. Some web server implementations may be unable to provide
+  this (e.g. if TLS is terminated by a separate proxy or load balancer); in that
+  case this shall be ``None``. If ``client_cert_chain`` is provided and
+  non-empty then this field must be provided and must contain information that
+  is consistent with ``client_cert_chain[0]``.  Note that under some setups,
+  (e.g. where TLS is terminated by a separate proxy or load balancer and that
+  device forwards the client certificate name to the web server), this field
+  may be set even where ``client_cert_chain`` is not set.  Optional; if missing
+  defaults to ``None``.
+
+* ``client_cert_error`` (*Optional[Unicode string]*) -- ``None`` if a client
+  certificate was provided and successfully verified, or was not provided.
+  If a client certificate was provided but verification failed, this is a
+  non-empty string containing an error message or error code indicating why
+  validation failed; the details are web server specific. Most web server
+  implementations will reject the connection if the client certificate
+  verification failed, instead of setting this value.  However, some may be
+  configured to allow the connection anyway.  This is especially useful when
+  testing that client certificates are supported properly by the client - it
+  allows a response containing an error message that can be presented to a
+  human, instead of just refusing the connection. Optional; if missing defaults
+  to ``None``.
+
+* ``tls_version`` (*Optional[int]*) -- The TLS version in use.  This is one of
+  the version numbers as defined in the TLS specifications, which is an
+  unsigned integer.  Common values include ``0x0303`` for TLS 1.2 or ``0x0304``
+  for TLS 1.3.  If TLS is not in use, set to ``None``.  Some web server
+  implementations may be unable to provide this (e.g. if TLS is terminated by a
+  separate proxy or load balancer); in that case set to ``None``. Mandatory.
+
+* ``cipher_suite`` (*Optional[int]*) -- The TLS cipher suite that is being
+  used.  This is a 16-bit unsigned integer that encodes the pair of 8-bit
+  integers specified in the relevant RFC, in network byte order.  For example
+  RFC8446 defines that the cipher suite ``TLS_AES_128_GCM_SHA256`` is
+  ``{0x13, 0x01}``; that is encoded as a ``cipher_suite`` value of
+  ``0x1301`` (equal to 4865 decimal).  Some web server implementations may be
+  unable to provide this (e.g. if TLS is terminated by a separate proxy or load
+  balancer); in that case set to ``None``. Mandatory.
+
+Events
+------
+
+All events are as defined in the *base protocol specification*.
+
+Rationale (Informative)
+-----------------------
+
+This section explains the choices that led to this specification.
+
+Providing the entire TLS certificates in ``client_cert_chain``, rather than a
+parsed subset:
+
+* makes it easier for webservers to implement, as they do not have to
```

Comment:
Nit: Let's capitalise the start of each bullet point

Reply:
Done!